spark/src/main/java/org/apache/comet/CometBatchIterator.java

```diff
@@ -33,12 +33,27 @@
 public class CometBatchIterator {
   final Iterator<ColumnarBatch> input;
   final NativeUtil nativeUtil;
+  private ColumnarBatch currentBatch = null;
 
   CometBatchIterator(Iterator<ColumnarBatch> input, NativeUtil nativeUtil) {
     this.input = input;
     this.nativeUtil = nativeUtil;
   }
 
+  /**
+   * Fetch the next input batch.
+   *
+   * @return Number of rows in next batch or -1 if no batches left.
+   */
+  public int hasNext() {
+    if (input.hasNext()) {
+      currentBatch = input.next();
```

Comment:
Hmmm this will advance `input` just by calling `hasNext`, will be a problem when this gets called twice?